cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -136,59 +138,73 @@ private MacroMetadata getMacroFunctionMetadata(int startIndex, int endIndex, Str
 
   private int getStartIndex(String str) {
     int startIndex = str.lastIndexOf("${");
-    while (escaped(startIndex, str)) {
-      // TODO: Break if doubly-escaped syntax
+    while (isEscaped(startIndex, str)) {
       startIndex = str.substring(0, startIndex - 1).lastIndexOf("${");
     }
     return startIndex;
   }
 
   private int getEndIndex(String str, int startIndex) {
     int endIndex = str.indexOf('}', startIndex);
-    while (escaped(endIndex, str)) {
-      // TODO: Break if doubly-escaped syntax
+    while (isEscaped(endIndex, str)) {
```

Comment:
Thanks, I forgot to do this as the three methods started off all a bit different... Will fix.